instructions/humus-common.instructions.md

```diff
@@ -0,0 +1,259 @@
+
+## Overview
+
+Humus is a modular Go framework built on [Bedrock](https://github.com/z5labs/bedrock) for creating production-ready REST APIs, gRPC services, jobs, and queue processors. Every application automatically includes OpenTelemetry instrumentation, health monitoring, and graceful shutdown.
+
+## Configuration
+
+**Always embed the appropriate Config type:**
+```go
+type Config struct {
+    rest.Config `config:",squash"`  // For REST services
+    // OR
+    grpc.Config `config:",squash"`  // For gRPC services
+    // OR
+    humus.Config `config:",squash"` // For Job/Queue services
+    
+    // Service-specific configuration
+    Database struct {
+        URL string `config:"url"`
+    } `config:"database"`
+}
+```
+
+**Implement provider interfaces when needed:**
+```go
+func (c Config) Listener(ctx context.Context) (net.Listener, error) {
+    return net.Listen("tcp", fmt.Sprintf(":%d", c.HTTP.Port))
+}
+```
+
+**Use Go templates in config.yaml:**
+```yaml
+openapi:
+  title: {{env "SERVICE_NAME" | default "My Service"}}
+  version: {{env "VERSION" | default "v1.0.0"}}
+
+database:
+  url: {{env "DATABASE_URL" | default "postgres://localhost/mydb"}}
+```
+
+## Error Handling
+
+**Runner-Level (applies to all service types):**
+```go
+runner := humus.NewRunner(builder, humus.OnError(humus.ErrorHandlerFunc(func(err error) {
+    log.Fatal(err)
+})))
+```
+
+**Operation-Level (REST-specific, see humus-rest.instructions.md):**
+```go
+rest.Handle(method, path, handler,
+    rest.OnError(rest.ErrorHandlerFunc(func(ctx context.Context, w http.ResponseWriter, err error) {
+        w.WriteHeader(http.StatusInternalServerError)
+        json.NewEncoder(w).Encode(map[string]string{"error": err.Error()})
+    })),
+)
+```
+
+## Logging & Observability
+
+**Always use humus.Logger:**
+```go
+log := humus.Logger("service-name")
+log.Info("user created", slog.String("user_id", userID))
+log.Error("failed to process", slog.String("error", err.Error()))
+```
+
+**Logs automatically correlate with OpenTelemetry traces** - no manual instrumentation needed.
+
+## Health Monitoring
+
+**Binary Health Check:**
+```go
+monitor := new(health.Binary)
+monitor.MarkHealthy()    // Service is healthy
+monitor.MarkUnhealthy()  // Service is unhealthy
+```
+
+**Composite Monitors:**
+```go
+// Both must be healthy
+health.And(dbMonitor, cacheMonitor)
+
+// At least one must be healthy
+health.Or(replica1Monitor, replica2Monitor)
+```
+
+## Lifecycle Management
+
+**Use lifecycle hooks for resource cleanup:**
+```go
+func Init(ctx context.Context, cfg Config) (*rest.Api, error) {
+    db, err := sql.Open("postgres", cfg.DB.URL)
+    if err != nil {
+        return nil, err
+    }
+    
+    lc, _ := lifecycle.FromContext(ctx)
+    lc.OnPostRun(lifecycle.HookFunc(func(ctx context.Context) error {
+        return db.Close()
+    }))
+    
+    return api, nil
+}
+```
+
+## Testing Patterns
```

Comment:
this more general go stuff and shouldn't be included in these instructions since users should already have instructions for go itself